src/Hl7.Fhir.Core/Serialization/SerializationUtil.cs

```diff
@@ -0,0 +1,485 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Xml.Linq;
+using System.Xml.Schema;
+using System.Xml;
+using System;
+using System.Text.RegularExpressions;
+using System.Text;
+using Hl7.Fhir.Utility;
+
+namespace Hl7.Fhir.Serialization
+{
+    public class SerializationUtil
+    {
+        public static bool ProbeIsXml(string data)
+        {
+            Regex xml = new Regex("^<[^>]+>");
+
+            return xml.IsMatch(data.TrimStart());
+        }
+
+        public static bool ProbeIsJson(string data) => data.TrimStart().StartsWith("{");
```

Comment:
This could also be a regex instead to avoid the possible allocation: `^\s*{`

Reply:
Done